### PR TITLE
Update syn related dependencies

### PR DESCRIPTION
**Pull request by Eijebong**

### CODE REVIEW

File: diesel_derives/Cargo.toml [hunk 1]
```diff
@@ -10,9 +10,9 @@ repository = "https://github.com/diesel-rs/diesel/tree/master/diesel_derives"
 autotests = false
 
 [dependencies]
-syn = { version = "0.15.0", features = ["full", "fold"] }
-quote = "0.6.0"
-proc-macro2 = "0.4.0"
+syn = { version = "1", features = ["full", "fold"] }
+quote = "1"
+proc-macro2 = "1"
```
Comment (JohnTitor) [inline diesel_derives/Cargo.toml]: IMO: these requirements' range is a bit wide. Should be `1.0`?

Comment (Eijebong) [reply diesel_derives/Cargo.toml]: It would do the same thing. 1.0 will happily pull 1.1.0 since they're semver compatible.

Comment (JohnTitor) [reply diesel_derives/Cargo.toml]: Ah, I forgot it! Looks good.

**Comment by theduke** [comment]
Any chance this can be pushed over the line?

**Comment by weiznich** [comment]
@theduke I've currently not the time to do this. It's probably the same for other diesel contributors, so if someone really want's this: submit a new PR with fixed compile test errors + run rustfmt on the changes.

**Comment by theduke** [comment]
I was hoping for @JohnTitor  but otherwise I could probably find the time this weekend.

**Comment by JohnTitor** [comment]
@theduke @weiznich I've opened a follow-up PR #2164

**Comment by weiznich** [comment]
Closed by #2164

**Pull request closed without merging**